File: 20221005/모범답안/20221005_08/GreatestLetterInBoth_test.js
import assert from 'assert'
import solution from './GreatestLetterInBoth.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'acBfbsTES'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('S', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'aA'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('A', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'aAzZ'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('Z', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'iSklTpxVxi'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'vxofFuUNALUWDZUhtgWO'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('U', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'FfbKycSNMnzjJBQKOmxL'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('N', result)
    done();
  });
})

File: 20221005/모범답안/20221005_08/GreatestLetterInBoth_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.GreatestLetterInBoth.GreatestLetterInBoth import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'acBfbsTES'

        # when
        result = solution(s)

        # then
        self.assertEqual('S', result)

    def test_2(self):
        # given
        s = 'aA'

        # when
        result = solution(s)

        # then
        self.assertEqual('A', result)

    def test_3(self):
        s = 'aAzZ'

        # when
        result = solution(s)

        # then
        self.assertEqual('Z', result)

    def test_4(self):
        s = 'iSklTpxVxi'

        # when
        result = solution(s)

        # then
        self.assertEqual('', result)

    def test_5(self):
        s = 'vxofFuUNALUWDZUhtgWO'

        # when
        result = solution(s)

        # then
        self.assertEqual('U', result)

    def test_6(self):
        s = 'FfbKycSNMnzjJBQKOmxL'

        # when
        result = solution(s)

        # then
        self.assertEqual('N', result)


if __name__ == '__main__':
    unittest.main()
